crates/oxc_syntax/src/reference.rs

```diff
@@ -11,6 +11,14 @@ use crate::{node::NodeId, symbol::SymbolId};
 #[derive(Debug, Clone, Copy, Eq, PartialEq, Ord, PartialOrd, Hash)]
 pub struct ReferenceId(NonMaxU32);
 
+impl<'alloc> CloneIn<'alloc> for ReferenceId {
+    type Cloned = Self;
+
+    fn clone_in(&self, _: &'alloc oxc_allocator::Allocator) -> Self::Cloned {
+        *self
+    }
+}
+
```

Comment:
```suggestion
impl<'alloc> CloneIn<'alloc> for ReferenceId {
    type Cloned = Self;

    fn clone_in(&self, _: &'alloc Allocator) -> Self {
        // `clone_in` should never reach this, because `CloneIn` skips semantic ID fields
        unreachable!();
    }

    fn clone_in_with_semantic_ids(&self, _: &'alloc Allocator) -> Self {
        *self
    }
}
```

(Please add `use oxc_allocator::Allocator;` with other imports. Our code style generally avoids local imports)